### DISCUSSION TITLE
OpenApi docs for subclass of list of pydantic BaseModel

### DISCUSSION THREAD

**Discussion by kosciej16** [fastapi/fastapi#6813] [Questions] [answered]
```Python
from typing import List
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class MyModel(BaseModel):
    a: int

class MyModelList(List[MyModel]):
    ...
    # Bunch of useful methods

# version 1. This one has working openapi docs
@app.get("/with_docs", response_model=List[MyModel])
def foo():
    return [MyModel(a=10)]

# version 2. How to make docs here works?
@app.get("/without_docs", response_model=MyModelList)
def foo2():
    res = MyModelList()
    res.append(MyModel(a=10)]
    return res
```

### Description
I search for a way to generate docs for type that is subclass of list of pydantic BaseModel. I have a bunch of usefull methods that operate on that list that's why I don't want to use version 1. I tried multiinheritance and other things but nothing worked.

**Comment by kosciej16** [comment, marked as answer]
Actually I have been enlightened during writing a question and found a solution, but spend some time on it so decided to share it with everyone.

We can just mix both versions.
```
@app.get("/without_docs", response_model=List[MyModel])
def foo2():
    res = MyModelList()
    res.append(MyModel(a=10)]
    return res
```

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
